app/models/concerns/dirty_associations.rb

```diff
@@ -14,16 +14,35 @@
 module DirtyAssociations
   extend ActiveSupport::Concern
 
+  included do
+      after_save :reset_dirty_cache_state
```

Comment:
I don't like this change tbh, this module is always expected to be included in active record model, which always has after_save defined. It's just the test that we wanted to keep simple and therefore we don't use the model there.

I was hoping to actually modify the test model so that it would define `after_save` even though it does not inherit from ActiveRecord. But if that does not feel right, I'd prefer the original version. Hence, please drop this commit from the PR and I'll merge. Sorry for going back and forth.